src/js/components/Anchor/StyledAnchor.js

```diff
@@ -24,7 +24,13 @@ const sizeStyle = props => {
   `;
 };
 
-const StyledAnchor = styled.a`
+const StyledAnchor = styled.a.withConfig({
+  // prevent custom props from bleeding into DOM
+  // https://styled-components.com/docs/api#shouldforwardprop
+  shouldForwardProp: (prop, defaultValidatorFn) =>
+    !['as', 'colorProp', 'hasIcon', 'hasLabel'].includes(prop) &&
+    defaultValidatorFn(prop),
+})`
```

Comment:
I wonder how come we are not seeing those props leaking on the tests/snapshots.

Reply:
I think the issue has to do when the components are being composed together and they share prop names. In this case, @mihai-ro was [trying to use Anchor + React-router Link](https://grommet.slack.com/archives/C09QQEG69/p1615558841146600).

We have also run into this with [StyledText](https://github.com/grommet/grommet/issues/4003#issuecomment-656820442) and [StyledButtonKind](https://github.com/grommet/grommet/blob/c65701c11c791169e5be90a330e41d5ad9f61f3e/src/js/components/Button/StyledButtonKind.js#L214).

Documentation: https://styled-components.com/docs/api#shouldforwardprop